app/src/test/java/pm/gnosis/heimdall/ui/tokens/AddTokenViewModelTest.kt

```diff
@@ -0,0 +1,108 @@
+package pm.gnosis.heimdall.ui.tokens
+
+import io.reactivex.Observable
+import io.reactivex.observers.TestObserver
+import org.junit.Assert.assertEquals
+import org.junit.Assert.assertNull
+import org.junit.Before
+import org.junit.Rule
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.mockito.BDDMockito.given
+import org.mockito.BDDMockito.then
+import org.mockito.Mock
+import org.mockito.junit.MockitoJUnitRunner
+import pm.gnosis.heimdall.common.util.DataResult
+import pm.gnosis.heimdall.common.util.ErrorResult
+import pm.gnosis.heimdall.common.util.Result
+import pm.gnosis.heimdall.data.exceptions.InvalidAddressException
+import pm.gnosis.heimdall.data.repositories.TokenRepository
+import pm.gnosis.heimdall.data.repositories.model.ERC20Token
+import pm.gnosis.heimdall.test.utils.ImmediateSchedulersRule
+import pm.gnosis.heimdall.test.utils.MockUtils
+import pm.gnosis.heimdall.test.utils.TestCompletable
+import pm.gnosis.utils.asEthereumAddressString
+import java.math.BigInteger
+
+@RunWith(MockitoJUnitRunner::class)
+class AddTokenViewModelTest {
+    @JvmField
+    @Rule
+    val rule = ImmediateSchedulersRule()
+
+    @Mock
+    private lateinit var tokenRepositoryMock: TokenRepository
+
+    private lateinit var viewModel: AddTokenViewModel
+
+    private val testAddress = BigInteger.ZERO
+    private val testToken = ERC20Token(testAddress, decimals = 18)
+
+    @Before
+    fun setUp() {
+        viewModel = AddTokenViewModel(tokenRepositoryMock)
+    }
+
+    @Test
+    fun addToken() {
+        val testObserver = TestObserver<Result<Unit>>()
+        val testCompletable = TestCompletable()
+        given(tokenRepositoryMock.addToken(MockUtils.any())).willReturn(testCompletable)
+
+        viewModel.erc20Token = testToken
```

Comment:
I really think we should try to avoid this. You could mock the loadToken info to test the same, right?